lib/constructBodyDigestSignature.js

```diff
@@ -0,0 +1,60 @@
+const crypto = require('crypto')
```

Comment:
Ok so I've had a go through everything and I think we can simplify this down some more. 
Generally speaking, I think there are a few assumptions about how this should work. 
- This method should only calculate the digest value. (single responsibility)
- This method can can (and should be overridden if someone needs to change its implementation) 
- The default for this should assume that: 
  1) The digest only consists of the request body
  2) Body content is always JSON
  3) Digest value is located in the digest header (this is where we get the algorithm)

If we follow those rules, this method should be called `getDigest` and is the default option.

we should have a `digestEncoding` option (at the top level which tells us how the string is encoded)
getDigest should receive as params: the request object, and options

inside getDigest, we:

1) Pull the digest header out of the request
2) grab the algorithm from the digest header value
3) grab the request body
4) hash the body with the correct algorithm and encode it with value provided buy `digestEncoding`
5) we can do a quick compare here and see if the strings are the same
6) Err if not, otherwise, return digest string

This way we wont need to worry about all the other bits (multiple digests etc), if someone needs that, they can override the getDigest method at the top level and handle it themselves :) 

Does that make sense to you? Thats the way I view this at least.

Reply:
Makes sense to me. 

One clarification though: Should the method be passed the request and then pull the Digest header out itself or be passed the Digest Header directly?

Reply:
Great input! I've adjusted the PR according to this comment. I'll add documentation on the default `getDigest` assumptions in a later PR.